dashboards-reports/server/routes/lib/createReport.ts

```diff
@@ -90,6 +92,14 @@ export const createReport = async (
 
       const [value, release] = await semaphore.acquire();
       try {
+        if (reportParams.core_params.header !== '') {
+          reportParams.core_params.header = converter.makeHtml(reportParams.core_params.header);
+          reportParams.core_params.header = replaceBlockedKeywords(reportParams.core_params.header);
+        }
+        if (reportParams.core_params.footer !== '') {
+          reportParams.core_params.footer = converter.makeHtml(reportParams.core_params.footer);
+          reportParams.core_params.footer = replaceBlockedKeywords(reportParams.core_params.footer);
```

Comment:
don't know if this should go into the report definition, in the definition we should still store user raw input and only render it to html before displaying it. otherwise user's markdown input becomes html in the UI

Reply:
something like this in visual report helper?
```js
  const reportHeader = header
    ? replaceBlockedKeywords(DOMPurify.sanitize(converter.makeHtml(header)))
    : DEFAULT_REPORT_HEADER;
```

Reply:
in `createVisualReport` I don't think this entry is actually being pushed to the index. When I view the raw header in report details or definition details, it still displays the markdown.

Reply:
should replaceBlockedKeywords be after DOMPurify.sanitize? say i constructed a string 
`local<bad-html>host` and `<bad-html>` hits purify's pattern got removed, what's left is still `localhost`